File: mobile/src/types/calendar.ts
import { SessionType, SessionMode, SessionStatus } from './common';

export interface CalendarEventResponse {
  id: string;
  title: string;
  start: string;
  end: string;
  session_type: SessionType;
  mode: SessionMode;
  status: SessionStatus;
  role: string;
  tutor_name: string | null;
  location_city: string | null;
}

export interface CalendarDateRange {
  start_date: string;
  end_date: string;
}

File: backend/app/schemas/calendar.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class CalendarEventResponse(BaseModel):
    id: uuid.UUID
    title: str
    start: datetime
    end: datetime
    session_type: str
    mode: str
    status: str
    role: str
    tutor_name: str | None = None
    location_city: str | None = None

    model_config = {"from_attributes": True}
